infra_monitored_account/infra_monitored_account/infra_monitored_stack.py

```diff
@@ -89,48 +97,83 @@ def create_event_rules(self, cross_account_bus_role, cross_account_event_bus_arn
-        """ 
+        """
         # todo: do we need multiple arns (lambdas)?
         # todo: how to get tooling extract lambda role -> through AWSNaming?
-        tooling_extract_lambda_role_name = AWSNaming.IAMRole(self, CDKResourceNames.IAMROLE_EXTRACT_METRICS_LAMBDA)
-        principal_arn = AWSNaming.Arn_IAMRole(self, self.tooling_account_id, tooling_extract_lambda_role_name)
-        
+        tooling_extract_lambda_role_name = AWSNaming.IAMRole(
+            self, CDKResourceNames.IAMROLE_EXTRACT_METRICS_LAMBDA
+        )
+        principal_arn = AWSNaming.Arn_IAMRole(
+            self, self.tooling_account_id, tooling_extract_lambda_role_name
+        )
+
         # todo rename
         cross_account_iam_role_extract_metrics = iam.Role(
-            self, "MonitoredAccExtractMetricsRole",
-            role_name=AWSNaming.IAMRole(self, CDKResourceNames.IAMROLE_MONITORED_ACC_EXTRACT_METRICS),            
-            assumed_by=iam.ArnPrincipal(principal_arn)
+            self,
+            "MonitoredAccExtractMetricsRole",
+            role_name=AWSNaming.IAMRole(
+                self, CDKResourceNames.IAMROLE_MONITORED_ACC_EXTRACT_METRICS
+            ),
+            assumed_by=iam.ArnPrincipal(principal_arn),
         )
 
         # Glue Policy
         glue_policy_statement = iam.PolicyStatement(
             actions=[
                 "glue:ListJobs",
+                "glue:ListWorkflows",
                 "glue:GetJob",
                 "glue:GetJobs",
                 "glue:GetJobRun",
                 "glue:GetJobRuns",
             ],
             resources=["*"],
-            effect=iam.Effect.ALLOW
+            effect=iam.Effect.ALLOW,
+        )
+        glue_inline_policy = iam.Policy(
+            self, "glue-extract", policy_name=AWSNaming.IAMPolicy(self, "glue-extract")
         )
-        glue_inline_policy = iam.Policy(self, "glue-extract", policy_name=AWSNaming.IAMPolicy(self,"glue-extract"))
         glue_inline_policy.add_statements(glue_policy_statement)
 
         # Lambda Policy (extract via CloudWatch logs)
         lambda_policy_statement = iam.PolicyStatement(
             actions=[
+                "lambda:ListFunctions",
                 "logs:GetQueryResults",
                 "logs:StartQuery",
                 "logs:StopQuery",
             ],
             resources=["*"],
-            effect=iam.Effect.ALLOW
+            effect=iam.Effect.ALLOW,
+        )
+        lambda_inline_policy = iam.Policy(
+            self,
+            "lambda-extract",
+            policy_name=AWSNaming.IAMPolicy(self, "lambda-extract"),
         )
-        lambda_inline_policy = iam.Policy(self, "lambda-extract", policy_name=AWSNaming.IAMPolicy(self,"lambda-extract"))
         lambda_inline_policy.add_statements(lambda_policy_statement)
 
+        # Step Functions Policy
+        states_policy_statement = iam.PolicyStatement(
```

Comment:
Maybe, it's better to name it step_functions_policy_statement, step_functions_policy?

Reply:
Done